rotkehlchen/tests/api/test_assets.py

```diff
@@ -699,3 +742,40 @@ def test_search_assets_with_levenshtein(rotkehlchen_api_server):
     )
     result = assert_proper_response_with_result(response)
     assert len(result) == 0
+
+    # check that using chain filter works.
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchlevenshteinresource',
+        ),
+        json={
+            'value': 'dai',
+            'limit': 50,
+            'chain': 'ethereum',
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert 50 >= len(result) > 10
+    assert all(['ethereum' == entry['chain'] for entry in result])
+    assert_substring_in_search_result(result, 'dai')
+    # check that Dai(DAI) appears at the top of result.
+    assert_asset_at_top_position(
+        asset_id='eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F',
+        max_position_index=0,
+        result=result,
+    )
+
+    # check that using an unsupported chain fails
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'assetssearchlevenshteinresource',
+        ),
+        json={
+            'value': 'dai',
+            'limit': 50,
+            'chain': 'near',
+        },
+    )
+    assert_error_response(response, contained_in_msg='Failed to deserialize ChainID value near')
```

Comment:
Here you are repeating code and I believe there should be probably other things repeating in the tests. For tests that are common to both endpoints wouldn't make sense to parametrize the test? Then for special checks like checking if dai is at the top you can add an if statement. What do you think?

Reply:
it isn't really repeated as they both accept different request JSON body.